lib/game.rb

```diff
@@ -0,0 +1,78 @@
+# frozen_string_literal: true
+
+require_relative '../dependencies'
+
+class Game
+  include Output
+  SECRET_CODE_LENGTH = 4
+  RANGE_START = 1
+  RANGE_END = 6
+  NOT_YET = '-'
+  GOT_IT = '+'
+  DIFFICULTY_LEVEL = {
+    easy: { attempts: 15, hints: 3 },
+    medium: { attempts: 10, hints: 2 },
+    hell: { attempts: 5, hints: 1 }
+  }.freeze
+
+  attr_accessor :attempts_total, :attempts, :difficulty, :hints_total, :hints, :name, :win, :secret
+
+  def initialize(name:, difficulty:)
+    @name = name
+    @difficulty = difficulty
+    @attempts = DIFFICULTY_LEVEL[difficulty][:attempts]
+    @hints = DIFFICULTY_LEVEL[difficulty][:hints]
+    @win = false
+    @secret = make_number
+    @unused_hints = @secret.chars
+  end
+
+  def make_number(numbers = RANGE_END)
+    (1..SECRET_CODE_LENGTH).map { rand(RANGE_START..numbers) }.join
+  end
+
+  def check(number)
```

Comment:
this method should not return true/false
it should return a comparison result, for example '++--'

Reply:
assign the result to an instance variable `@last_result` and then in `win` method you can compare `@last_result` with `GOT_IT * SECRET_CODE_LENGTH`
no need for the `@win` variable, please remove it